### PR TITLE
feat: delete an activity log

### PR DESCRIPTION
**Pull request by Laban254**
This pr introduces a new endpoint for deleting activity logs. The changes include:

-   **Endpoint:** `DELETE /api/v1/activity-logs/{id}`
-   **Functionality:** Allows authenticated super admins to delete an activity log by its ID.
-   **Authorization:** Only accessible by super admins.
-   **Response Handling:** Returns appropriate status codes and messages based on the outcome (success, unauthorized, forbidden, not found, server error).

**Related Issue:**

[Issue 765](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/765) - Implement  delete activity log endpoint

**Motivation and Context:**

This feature is necessary to provide administrators with the capability to manage and clean up activity logs. It ensures that logs can be removed when no longer needed, which helps maintain data relevance and integrity.

**How Has This Been Tested?**

The delete activity log feature was tested by:

-   **Endpoint Testing:** Verified that the endpoint correctly deletes activity logs and handles various scenarios (success, unauthorized access, forbidden access, not found, server errors).
-   **Authorization Testing:** Ensured that only authenticated super admins can perform deletions.
-   **Error Handling:** Confirmed that appropriate error messages are returned for unauthorized access and non-existent logs.

**Screenshots:**

Screenshot from 2024-08-07 16-33-46

Screenshot from 2024-08-07 16-24-52

**Types of changes:**

- [ ]  Bug fix (non-breaking change which fixes an issue)
-    [x] New feature (non-breaking change which adds functionality)
-  [ ]  Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**

-    [x] My code follows the code style of this project.
-    [ ] My change requires a change to the documentation.
-   [ ]  I have updated the documentation accordingly.
-    [x] I have read the CONTRIBUTING document.
-   [x]  I have added tests to cover my changes.
-    [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/activity_logs.py [hunk 1]
```diff
@@ -70,4 +70,26 @@ async def fetch_all_users_activity_log(
...
+
+@activity_logs.delete("/{log_id}", status_code=status.HTTP_204_NO_CONTENT)
+async def delete_activity_log(
+    log_id: str,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(user_service.get_current_super_admin)
+):
+    """Endpoint to delete an activity log by its ID"""
+
+    try:
```
Comment (joboy-dev) [inline api/v1/routes/activity_logs.py:83]: No need fot this exception handling. It is already taken care of

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

File: api/v1/routes/activity_logs.py (hunk 1 above)
Comment (Laban254) [reply api/v1/routes/activity_logs.py:83]: ok, let me work on it

**Comment by Laban254** [comment]
> Make changes

@joboy-dev I have made the changes

**Review by joboy-dev — APPROVED** [review]

**Pull request merged**